File: keemsebin/5주차/1940.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);
  let M = Number(input[1]);
  let result = 0;

  let ingredient = input[2].split(" ").map(Number);

  for (let i = 0; i < ingredient.length; i++) {
    for (let j = i + 1; j < ingredient.length; j++) {
      if (ingredient[j] + ingredient[i] == M) {
        result++;
        break;
      }
    }
  }

  console.log(result);
});

File: k-kbk/5주차/1940.py
import sys

input = sys.stdin.readline

n = int(input())
m = int(input())
items = list(map(int, input().split()))

items.sort()
count = 0
start = 0
end = len(items) - 1

while start < end:
    value = items[start] + items[end]
    if value < m:
        start += 1
    elif value > m:
        end -= 1
    else:
        count += 1
        start += 1
        end -= 1

print(count)
